src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,252 @@
+            }
+        }
+    }
+    return total;
+};
+
+const calculateSum = (values): number => {
+    const numericVals = values.filter((x) => typeof x == "number");
+    return numericVals.length > 0 ? numericVals.reduce((s: number, a: number) => s + a, 0) : 0;
+};
+
+const calculateAverage = (values): number => {
+    const numericVals = values.filter((x) => typeof x == "number");
+    const sum = numericVals.length > 0 ? numericVals.reduce((s: number, a: number) => s + a, 0) : 0;
+    return sum / numericVals.length || 0;
+};
+/**
+ * A table form element.
+ * @displayName Editable Table
+ * @description A table form element that allows for inline editing and tabular calculation.
+ * @param props The props that will be provided by the Workflow runtime.
+ */
+function EditableTableElement(props: EditableTableElementProps): React.ReactElement | null {
+    const {
+        rows,
+        cols,
+        maxHeight,
+        maxWidth,
+        onMouseEnter,
+        onMouseLeave,
+        raiseEvent,
+        setValue,
+        size,
+        stickyHeader,
+        footerLabel,
+    } = props;
+    const [data, setData] = useState(rows);
+    setValue(data);
+    useEffect(() => {
+        setData(rows);
+    }, [rows]);
+
+    const illegalCols = cols.filter(
+        (x) => !!x.type && ["text", "number", "date", "time", undefined].indexOf(x.type) === -1
+    );
+    if (illegalCols.length > 0) {
+        for (const illegalCol of illegalCols) {
+            const type = illegalCol.type as string;
+            console.log(
+                `Unsupported column type ${type} in column ${illegalCol.name}.  Supported types: text, number.`
+            );
+            illegalCol.type = undefined;
+        }
+    }
+    const includeFooter = cols.some((x) => x.columnCalculation);
+    const footerRow: FooterRow = calculateFooterRow(cols, rows);
+
+    const handleMouseEnter = (event: React.MouseEvent, row: RowData) => {
+        raiseEvent("custom", {
+            eventType: "mouseEnter",
+            rowData: { rows: data, row: row },
+        });
+        onMouseEnter?.(row);
+    };
+
+    const handleMouseLeave = (event: React.MouseEvent, row: RowData) => {
+        raiseEvent("custom", {
+            eventType: "mouseLeave",
+            rowData: { rows: data, row: row },
+        });
+        onMouseLeave?.(row);
+    };
+
+    const handleCellChange = (
+        event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+        row: RowData,
+        column: Column
+    ) => {
+        if (column?.parse) {
+            row[column.name] = column.parse(event.target.value, row, column);
+        } else if (event.target.type === "number") {
+            row[column.name] = Number(event.target.value);
+        } else {
+            row[column.name] = event.target.value;
+        }
+        raiseEvent("custom", {
+            eventType: "cellChange",
+            rowData: { rows: data, row: row, column: column },
```

Comment:
I'd expect something more like this:
```
raiseEvent("custom", {
    eventType: "cellChange",
    row, 
    column,
    value: row[column.name],
});
```

We shouldn't need to provide the whole collection of rows.